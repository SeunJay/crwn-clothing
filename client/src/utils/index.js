import { applyMiddleware, createStore } from "redux";
import rootReducer from "../redux/root-reducer";
import { middlewares } from "../redux/store";

export const testStore = (initialState) => {
  const createStoreWithMiddleware = applyMiddleware(...middlewares)(
    createStore
  );
  return createStoreWithMiddleware(rootReducer, initialState);
};
