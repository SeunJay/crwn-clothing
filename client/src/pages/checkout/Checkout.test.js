import React from "react";
import { shallow } from "enzyme";
// import configureStore from "redux-mock-store"
import { testStore } from "../../utils";
import Checkout from "./Checkout";

// let wrapper;

const setUp = (initialState = {}) => {
  const store = testStore(initialState);
  const wrapper = shallow(<Checkout store={store} />);
  return wrapper;
};

describe("Checkout Page", () => {
  let wrapper;

  beforeEach(() => {
    const initialState = {
      cart: {
        cartItems: [],
        total: 50
      },
      
    };

    wrapper = setUp(initialState);
  });

  it("take snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render the Checkout component", () => {
    expect(wrapper.length).toBe(1);
  });

  
});
