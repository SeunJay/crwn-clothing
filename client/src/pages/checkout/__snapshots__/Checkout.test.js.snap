// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout Page  1`] = `ShallowWrapper {}`;

exports[`Checkout Page should render the Checkout component 1`] = `undefined`;

exports[`Checkout Page take snapshot 1`] = `ShallowWrapper {}`;
